crates/binstalk/src/fetchers/quickinstall.rs

```diff
@@ -21,9 +22,39 @@ use super::{Data, TargetData};
 const BASE_URL: &str = "https://github.com/cargo-bins/cargo-quickinstall/releases/download";
 const STATS_URL: &str = "https://warehouse-clerk-tmp.vercel.app/api/crate";
 
+const QUICKINSTALL_SUPPORTED_TARGETS_URL: &str =
+    "https://raw.githubusercontent.com/cargo-bins/cargo-quickinstall/main/supported-targets";
+
+async fn get_quickinstall_supported_targets(
+    client: &Client,
+) -> Result<&'static [CompactString], BinstallError> {
+    static SUPPORTED_TARGETS: OnceCell<Box<[CompactString]>> = OnceCell::const_new();
+
+    SUPPORTED_TARGETS
+        .get_or_try_init(|| async {
+            let bytes = client
+                .get(Url::parse(QUICKINSTALL_SUPPORTED_TARGETS_URL)?)
+                .send(true)
+                .await?
+                .bytes()
+                .await?;
+
+            let mut v: Vec<CompactString> = String::from_utf8_lossy(&bytes)
+                .split_whitespace()
+                .map(CompactString::new)
+                .collect();
+            v.sort_unstable();
+            v.dedup();
+            Ok(v.into())
+        })
+        .await
+        .map(Box::as_ref)
+}
```

Comment:
Hi, I'm the person who made the issue that this PR fixes.

I read this code, and I'm a little concerned that GitHub might not like that `supported-targets` is being obtained from the repo every time `cargo-binstall` is used? It's a small file, and `cargo-binstall` isn't being used 30 times a day from every computer that uses it, but... still. I have my concerns.

Reply:
If you're concerned by that, you'll be alarmed to learn about our strategy when there's no github token available, which hits github dozens or hundreds of times _per package_. GitHub's never complained! I think we're fine.

Reply:
...Oh. Well that's good to hear. Thank you for letting me know. :)

Reply:
> Hi, I'm the person who made the issue that this PR fixes.

Thanks for reporting the issue!

> I read this code, and I'm a little concerned that GitHub might not like that `supported-targets` is being obtained from the repo every time `cargo-binstall` is used? It's a small file, and `cargo-binstall` isn't being used 30 times a day from every computer that uses it, but... still. I have my concerns.

It should be ok, since it will be fetched once per `cargo-binstall` invocation.
`cargo-binstall` also issues lots of other http requests to http, so I don't think that is particularly concerning compared to that.

If it becomes a problem, I can try to cache it on disk.